cmd/soroban-cli/src/config/locator.rs

```diff
@@ -260,7 +262,23 @@ impl Args {
         res
     }
 
-    pub fn remove_identity(&self, name: &str) -> Result<(), Error> {
+    pub fn remove_identity(&self, name: &str, global_args: &global::Args) -> Result<(), Error> {
+        let print = Print::new(global_args.quiet);
+        let identity = self.read_identity(name)?;
+        if let Secret::SecureStore { entry_name } = identity {
+            let entry = StellarEntry::new(&entry_name)?;
+            match entry.delete_seed_phrase() {
+                Ok(()) => {}
+                Err(e) => match e {
+                    signer::keyring::Error::Keyring(keyring::Error::NoEntry) => {
+                        print.infoln("This key was already removed from the secure store. Removing the cli config file.");
+                    }
+                    _ => {
+                        return Err(Error::Keyring(e));
+                    }
+                },
+            }
+        }
         KeyType::Identity.remove(name, &self.config_dir()?)
```

Comment:
Not sure this will be an issue, but it seems like it could be possible for the keyring to successfully delete the key but not the Identity file? I think we can assume this isn't the case since we read the id first.

Reply:
🤔 I see what you're saying. I think you're right that since we are looking up the identity first, we know that the file path is correct, so it _should_ be able to delete cleanly. 

I wonder if its worth removing the identity file first. If that fails, then we return. And then if it succeeds, but the keychain removal fails, at least the identity won't show up in the CLI anymore. That's not ideal, but seems like a less confusing user experience.

Reply:
I'm going to resolve this so we can get this merged in, but happy to keep talking/thinking about it!